### PR TITLE
[PD-1919] Adding comma to locations.

### PR DESCRIPTION
**Pull request by misterjonscott**
Adding commas to location string in job detail templates, and tests to verify that they're being rendered appropriately.

### CODE REVIEW

File: seo/tests/test_views.py [hunk 1]
```diff
@@ -1715,6 +1715,81 @@ def test_title_boost(self):
         test_search_not(self)
         test_title_boost(self)
 
+    def test_job_detail_heading(self):
+        """
+        Test to make sure job location is printed with commas, appropriately.
+        """
+        SeoSite.objects.get(id=1).delete()
+        ats = factories.ATSSourceCodeFactory.build()
```
Comment (aspidites) [inline seo/tests/test_views.py]: The save lines after *Factory.build() are redundant. Actually, build is redundant, as it's the default build strategy, so *Factory() is sufficient. In this case: `ats = factories.ATSSOurceCodeFactory()`

File: seo/tests/test_views.py [hunk 2]
```diff
@@ -1715,6 +1715,81 @@ def test_title_boost(self):
...
+        ats.save()
+        gac = factories.GACampaignFactory.build()
+        gac.save()
+        site = factories.SeoSiteFactory.build(
+            google_analytics_campaigns=gac,
+            view_sources=factories.ViewSourceFactory(id=1),
+            id=1)
+        site.save()
+        site.ats_source_codes.add(ats),
+        site.special_commitments.add(factories.SpecialCommitmentFactory(id=1))
```
Comment (aspidites) [inline seo/tests/test_views.py]: `save()` is implicit after an `add()` operation, thus redundant.

**Comment by misterjonscott** [comment]
Updated with shortened lines, and changes based on @aspidites feedback.

File: seo/tests/test_views.py [hunk 3]
```diff
@@ -1715,13 +1714,89 @@ def test_title_boost(self):
...
+        view_source = factories.ViewSourceFactory()
+        special_commitment = factories.SpecialCommitmentFactory()
+        
+        # Job lookup by guid, check for full location.
+        resp = self.client.get(
+            u'/indianapolis-in/retail-associate-розничная-ассоциированных/11111111111111111111111111111111/job/',
+            HTTP_HOST='buckconsultants.jobs')
+        self.assertEqual(resp.status_code, 200)
+        self.assertContains(resp, 'itemtype="http://schema.org/PostalAddress">'\
```
Comment (tdphillips) [inline seo/tests/test_views.py]: You probably need to compare against `resp.content`

Comment (JLMcLaugh) [inline seo/tests/test_views.py]: assertContains takes the response as the first argument, not a string.

Comment (misterjonscott) [inline seo/tests/test_views.py]: I'm confused about what's being requested, here.  Are there changes required?  This was tested.

File: seo/tests/test_views.py [hunk 4]
```diff
@@ -1715,13 +1714,89 @@ def test_title_boost(self):
...
+                                  '<span itemprop="addressCountry">United States</span></span>')
+
     def test_job_detail(self):
         """
         Test that job detail pages return no server errors and that objects
         all work.
 
         """
-        SeoSite.objects.get(id=1).delete()
```
Comment (JLMcLaugh) [inline seo/tests/test_views.py]: This line needs to go back in.

Comment (misterjonscott) [inline seo/tests/test_views.py]: I've added it back.

File: seo/tests/test_views.py [hunk 5]
```diff
@@ -1759,6 +1834,7 @@ def test_job_detail(self):
             u'/indianapolis-in/retail-associate-розничная-ассоциированных/11111111111111111111111111111111/job/',
             HTTP_HOST='buckconsultants.jobs')
         self.assertEqual(resp.status_code, 200)
+        self.assertContains(resp, 'Indianapolis, Indiana, United States')
```
Comment (JLMcLaugh) [inline seo/tests/test_views.py]: I think this is matching the title, not the heading we're looking for. assertContains doesn't ignore html.

Comment (misterjonscott) [inline seo/tests/test_views.py]: I removed this line, as it's the one we added to the test for response code, and it's not part of the test for location formatting.

File: seo/tests/test_views.py (hunk 3 above)
Comment (aspidites) [inline seo/tests/test_views.py]: Same feedback I gave Daniel: I unit test should be testing a single unit of code. That there are three different scenarios we are testing here necessitates three different tests. I'm ok with testing a precondition though.

File: seo/tests/test_views.py [hunk 6]
```diff
@@ -1715,6 +1714,83 @@ def test_title_boost(self):
...
+        view_source = factories.ViewSourceFactory()
+        special_commitment = factories.SpecialCommitmentFactory()
+        
+        # Job lookup by guid, check for full location.
+        resp = self.client.get(
+            u'/indianapolis-in/retail-associate-розничная-ассоциированных/11111111111111111111111111111111/job/',
+            HTTP_HOST='buckconsultants.jobs')
+        self.assertEqual(resp.status_code, 200)
+        self.assertContains(resp, 'itemtype="http://schema.org/PostalAddress">'\
+                                  '<span itemprop="addressLocality">Indianapolis</span>, '\
```
Comment (aspidites) [inline seo/tests/test_views.py]: The `\`s are redundant. Strings within parens are automatically concatenated. Eg:

```python
# this
('foo bar '
  'baz buzz')
# becomes this
('foo bar baz buzz')
```

**Comment by aspidites** [comment]
Maybe I commented on the wrong ticket before? Thought I mentioned unit tests being independent, but I don't see that comment anymore. Oh well. I was just mentioning that different scenarios should have different test cases. As it stands now, if the empty city/state case fails, we have to get passed the others in order to iteratively fix/test it. Further, if the one test case fails, it isn't obvious which of the cases failed until you examine the test case.

**Comment by misterjonscott** [comment]
Added messages to subsection tests within heading test, to make review of failures easier.

**Comment by misterjonscott** [comment]
The change from None to '' didn't help.  We've been unable to see what's happening differently in Jenkins' build vs local dev.  Still working.

**Comment by misterjonscott** [comment]
Indiana is still being rendered into the page, in the Jenkins build, where it's not in the local build.  That's why this test is breaking...  Unsure why that would happen.

File: seo/tests/test_views.py [hunk 7]
```diff
@@ -1715,6 +1715,89 @@ def test_title_boost(self):
...
+        }
+        solr_docs_copy[0].update(kwargs1)
+        self.conn.add(solr_docs_copy)
+        resp = self.client.get(
+            u'/indianapolis-in/retail-associate-розничная-ассоциированных/11111111111111111111111111111111/job/',
+            HTTP_HOST='buckconsultants.jobs')
+        # import ipdb; ipdb.set_trace()
+        self.assertEqual(resp.status_code, 200, msg="Not receiving 200 response in test for no state")
+        self.assertContains(resp, 'itemtype="http://schema.org/PostalAddress">'
+                                  '<span itemprop="addressLocality">Indiandapolis</span>'
```
Comment (tdphillips) [inline seo/tests/test_views.py]: Should this contain a comma? According to Jenkins, the response contains a comma.

**Comment by aspidites** [comment]
What ended up being the fix?

**Comment by tdphillips** [comment]
Removing the tests. :/

**Comment by aspidites** [comment]
That's... Less than satisfactory

**Pull request merged**